src/odemis/acq/align/z_localization.py

```diff
@@ -300,6 +315,30 @@ def _do_measure_z(f: Future, stigmator, angle: float, pos: Tuple[float, float],
         logging.debug("Located feature Z shifted by %s m", zshift)
 
         return zshift, warning
+
+    except CancelledError:
+        raise
+
     finally:
-        f = stigmator.moveAbs({"rz": 0})
-        f.result(timeout=600)
+        f.running_subf = stigmator.moveAbs({"rz": 0})
+        f.running_subf.result(timeout=600)
+        with f._task_lock:
+            if f._task_state == CANCELLED:
```

Comment:
Do you want the final sub-future task `stigmator.moveAbs({"rz": 0})` to handle cancel and raise an exception? or this final sub-future should always be executed?